common/flatpak-utils-private.h

```diff
@@ -812,6 +812,18 @@ typedef void (*FlatpakProgressCallback)(const char *status,
 OstreeAsyncProgress *flatpak_progress_new (FlatpakProgressCallback progress,
                                            gpointer                progress_data);
 
+OstreeAsyncProgress *flatpak_progress_chain (OstreeAsyncProgress *progress);
+
+static inline void
+flatpak_progress_unchain (OstreeAsyncProgress *chained_progress)
+{
+  if (chained_progress != NULL)
+    ostree_async_progress_finish (chained_progress);
+}
+
+typedef OstreeAsyncProgress OstreeAsyncProgressChained;
```

Comment:
I think this should probably be `FlatpakAsyncProgressChained` (or something similar in the `Flatpak` namespace) so that you don’t wander into the `Ostree` namespace. That could cause issues if this functionality were to make its way into libostree in future, for example.